src/routes/newEntry/newEntry.component.tsx

```diff
@@ -54,133 +54,191 @@ export const NewEntry = () => {
     handleSubmit,
     formState: { errors },
   } = useForm<NewEntryInputs>();
-
   const onSubmit: SubmitHandler<NewEntryInputs> = async (data) => {
     setLoading(true);
-    const { icon } = data;
-    let iconId: string | undefined = undefined;
+    setIsError(false);
+    try {
+      const postIcon = async (icon: NewEntryInputs['icon']) => {
+        if (icon) {
+          const { data } = await postImage(icon);
+          return data.fileId;
+        }
+      };
+      const iconId = await postIcon(data.icon);
+      const entry = prepareNewEntry(data, iconId);
 
-    if (icon?.length) {
-      const { data } = await postImage(icon[0]);
-      iconId = data.fileId;
+      await postEntry(entry);
+      alert('Entry created!');
```

Comment:
Maybe we add a sticker component instead?

Reply:
I think I would create a separate task for that, as it will take some time